src/main/java/org/haffson/adventofcode/days/day07/StepSorter.java

```diff
@@ -0,0 +1,56 @@
+package org.haffson.adventofcode.days.day07;
+
+import javax.annotation.Nonnull;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.List;
+
+import static java.util.Objects.requireNonNull;
+
+public class StepSorter {
+    @Nonnull
+    private final SortedSteps sortedSteps;
+
+    public StepSorter() {
+        this.sortedSteps = new SortedSteps();
+    }
+
+    @Nonnull
+    public String getSortedSteps(@Nonnull final List<Step> stepsInput) {
```

Comment:
Wenn hier wieder `List<Step>` rauskommen würden, dann wären die Verantwortlichkeiten noch besser getrennt.

So sortiert die Klasse nicht nur, sondern bringt die Steps auch in ein String-Format.

Alternative: Steps-Klasse zurückgeben, die eine Liste von Steps kapselt. Da könnte man `toString()``auch gut unterbringen.

Reply:
Die Idee finde ich gut.
Habe mir hierzu verschiedene Sachen angeschaut und ausprobiert.
Schlussendlich lasse ich mir jetzt `sortedSteps` zurückgeben - die haben ja als Attribut eine List<Step>.
Hier lassen sich dann auch gut alle Hilfsfunktionen ansiedeln, wie `asCharacters`, oder `asString`.